Sources_v3/ChatClient_Tests.swift

```diff
@@ -143,6 +143,32 @@ class ChatClient_Tests: StressTestCase {
         XCTAssert(webSocket?.init_requestEncoder.connectionDetailsProviderDelegate === client)
     }
     
+    func test_webSocketClient_hasAllMandatoryMiddlewares() throws {
+        // Use in-memory store
+        // Create a new chat client, which in turn creates a webSocketClient
+        _ = ChatClient(
+            config: inMemoryStorageConfig,
+            workerBuilders: workerBuilders,
+            environment: testEnv.environment
+        )
+        
+        // Assert that mandatory middlewares exists
+        let middlewares = try XCTUnwrap(testEnv.webSocketClient?.init_eventNotificationCenter.middlewares)
+        // We currently have 4 mandatory middlewares, update this if you add more
+        // in `Client.eventNotificationCenter` lazy variable
+        // (3 of them added in `eventNotificationCenter` init, 1 added in `WebSocketClient` init)
+        XCTAssertEqual(middlewares.count, 4)
+        
+        // Assert `HealthCheckMiddleware` exists
+        XCTAssert(middlewares.contains(where: { $0 is HealthCheckMiddleware }))
+        // Assert `EventDataProcessorMiddleware` exists
+        XCTAssert(middlewares.contains(where: { $0 is EventDataProcessorMiddleware<DefaultDataTypes> }))
+        // Assert `TypingStartCleanupMiddleware` exists
+        XCTAssert(middlewares.contains(where: { $0 is TypingStartCleanupMiddleware<DefaultDataTypes> }))
+        // Assert `ChannelReadUpdaterMiddleware` exists
+        XCTAssert(middlewares.contains(where: { $0 is ChannelReadUpdaterMiddleware<DefaultDataTypes> }))
```

Comment:
This is enough for the test imo

Reply:
But if we don't assert the count, newly added middlewares won't be caught (just as I've added ReadUpdater middleware but forgot to add it to the list) ?

Reply:
I see your point. OK then :) I'd adjust the comment to exactly what you just said to me + I'd remove the `HealthCheckMiddleware` check here

Reply:
@VojtaStavik on a second thought, you're right. Asserting count is not needed, since if someone forgot to add the middleware, count will still hold and test will not warn them. Only way would be to check conformers of `EventMiddleware` protocol and make sure each conformer is added to the list of middlewares, which might be a bit overkill